client/db/bolt/db.go

```diff
@@ -818,18 +819,44 @@ func newestBuckets(master *bbolt.Bucket, n int, timeKey []byte, filter func(*bbo
 	return idx.pairs
 }
 
-// makeTopLevelBuckets creates a top-level bucket for each of the provided keys,
-// if the bucket doesn't already exist.
-func (db *boltDB) makeTopLevelBuckets(buckets [][]byte) error {
-	return db.Update(func(tx *bbolt.Tx) error {
+// init sets the database version and creates top-level bucket
+// for each of the provided keys, if the bucket doesn't already exist.
+func (db *boltDB) init(buckets [][]byte) error {
+	err := db.Update(func(tx *bbolt.Tx) error {
 		for _, bucket := range buckets {
 			_, err := tx.CreateBucketIfNotExists(bucket)
 			if err != nil {
 				return err
 			}
 		}
+
+		// Set the db version if it does not already exist.
+		bucket := tx.Bucket(appBucket)
+		if bucket == nil {
+			return fmt.Errorf("app bucket not found")
+		}
+		versionB := bucket.Get(versionKey)
+		if versionB == nil {
+			versionB = encode.Uint32Bytes(uint32(initialVersion))
```

Comment:
Why create a new db at version `initialVersion` instead of the current version if this is clearly a fresh DB?

Reply:
Thought it'd be ideal to run through all the upgrades for a database just in case a db that doesn't have its version set is provided.

Reply:
This was because of me. My comments are up a little further, but the gist was, dcrwallet always starts with the initial version. I don't know why, I can ask @jrick , unless it is obvious?

The only think I can think of, is as time goes on the initial state is forgotten. So you have a bunch of upgrade functions, but no reference as to what they are upgrading (other than comments). If they work perfect forever, I guess that's fine, but if a problem comes out later with an upgrade function that was written a year ago, like if a golang update breaks it, I would think it would be difficult to track down the problem, and impossible to test without knowing the state of databases needing that upgrade.

Reply:
I asked on matrix and @jrick had this to say:

>so there is only a single codepath to creating the db at that version
>otherwise it may happen that a db is different if it was upgraded vs created then

Which makes sense to me, and is in-line with what I thought. If you cater the code to only new databases, the upgrade functions become increasingly distanced.

Reply:
dcrd, however just starts at the current version. So, I guess that works too. However, dcrd only works with static data, which I think is an important distinction.